src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/repository/ActionPlanRepositoryTest.kt

```diff
@@ -0,0 +1,52 @@
+package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.repository
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.autoconfigure.orm.jpa.TestEntityManager
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.ActionPlan
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.SampleData
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.util.AuthUserFactory
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.util.RepositoryTest
+
+@RepositoryTest
+class ActionPlanRepositoryTest @Autowired constructor(
+  val entityManager: TestEntityManager,
+  val actionPlanRepository: ActionPlanRepository,
+) {
+  private val authUserFactory = AuthUserFactory(entityManager)
+
+  @Test
+  fun `can retrieve an action plan`() {
+    val actionPlan = buildAndPersistActionPlan()
+    val actionPlanFirstActivity = actionPlan.activities.first()
+
+    val savedPlan = actionPlanRepository.findById(actionPlan.id)
+
+    assertThat(savedPlan.get().numberOfSessions).isEqualTo(actionPlan.numberOfSessions)
+    assertThat(savedPlan.get().activities.size).isEqualTo(1)
+
+    val savedPlanFirstActivity = savedPlan.get().activities.first()
+    assertThat(actionPlanFirstActivity.desiredOutcome.id).isEqualTo(savedPlanFirstActivity.desiredOutcome.id)
+    assertThat(actionPlanFirstActivity.description).isEqualTo(savedPlanFirstActivity.description)
+  }
+
+  private fun buildAndPersistActionPlan(): ActionPlan {
```

Comment:
not for this PR - the factory methods aim to reduce the amount of code like this we have to write by chaining the repository persistence so you can just call one factory method and everything is guaranteed to be persisted. 

i've changed them in another PR so that you can use them in plain unit tests too (by just not passing an entity manager to the constructor).

Reply:
Understood